internal/build/buildconfig/manager.go

```diff
@@ -0,0 +1,187 @@
+package buildconfig
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/auth"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/build"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/registry"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const builderServiceAccountName = "builder"
+
+var (
+	ErrNoMatchingBuildConfig = errors.New("no matching BuildConfig")
+	errNoMatchingBuild       = errors.New("no matching Build")
+)
+
+type buildConfigManager struct {
+	authFactory     auth.RegistryAuthGetterFactory
+	client          client.Client
+	coreClient      kubernetes.Interface
+	helper          build.Helper
+	maker           Maker
+	ocpBuildsHelper OpenShiftBuildsHelper
+	registry        registry.Registry
+}
+
+func NewManager(
+	client client.Client,
+	coreClient kubernetes.Interface,
+	registry registry.Registry,
+	authFactory auth.RegistryAuthGetterFactory,
+	maker Maker,
+	ocpBuildsHelper OpenShiftBuildsHelper,
+	helper build.Helper,
+) *buildConfigManager {
+	return &buildConfigManager{
+		authFactory:     authFactory,
+		client:          client,
+		coreClient:      coreClient,
+		helper:          helper,
+		maker:           maker,
+		ocpBuildsHelper: ocpBuildsHelper,
+		registry:        registry,
+	}
+}
+
+func (bcm *buildConfigManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1beta1.KernelMapping, targetKernel string) (build.Result, error) {
+	logger := log.FromContext(ctx)
+
+	config := bcm.helper.GetRelevantBuild(mod, m)
+
+	var registryAuthGetter auth.RegistryAuthGetter
+
+	if irs := mod.Spec.ImageRepoSecret; irs != nil {
+		logger.Info("Getting pull secret from Kubernetes secret", "secret name", irs.Name)
+
+		namespacedName := types.NamespacedName{
+			Name:      irs.Name,
+			Namespace: mod.Namespace,
+		}
+
+		registryAuthGetter = bcm.authFactory.NewRegistryAuthGetter(bcm.client, namespacedName)
+	} else {
+		logger.Info("Getting pull secret from Service Account", "service account", builderServiceAccountName)
+
+		registryAuthGetter = bcm.authFactory.NewServiceAccountRegistryAuthGetter(bcm.coreClient, mod.Namespace, builderServiceAccountName)
+	}
+
+	imageAvailable, err := bcm.registry.ImageExists(ctx, m.ContainerImage, config.Pull, registryAuthGetter)
+	if err != nil {
+		return build.Result{}, fmt.Errorf("could not check if the image is available: %v", err)
+	}
+
+	if imageAvailable {
+		return build.Result{Status: build.StatusCompleted, Requeue: false}, nil
```

Comment:
WDYT about adding a new status `StatusSkipped` to `build.Result`? Using `StatusCompleted` doesn't feels right here.

Reply:
Why doesn't it feel right? If we add `StatusSkipped`, I believe we'll move straight from `build.StatusInProgress` to `build.StatusSkipped`.

Reply:
I may be missing something here but isn't that case cover the story in which the user has pre-build his driver? If that the case, how will we reach `build.STatusInProgress`?

Reply:
If the image is pre-built, then there won't be any `build` section for that mapping in the CRD and the reconciler will not call this method.

Reply:
So why do we check if the image is available to begin with? In which cases the image will already exist?

Reply:
@qbarrand i am not sure that we can count on build section not present in case image is prebuilt. Build section is a fallback. Maybe we need to move the image existence to an upper layer, prior to entering the build flow.  Mayb e it should be defined in the HandleBuild function in controller, or even before calling handleBuild